### PR TITLE
Completed Issue 3 User can edit an entry - Bryan Brown

### PR DESCRIPTION
**Pull request by bbrown949**
[ezgif.com-optimize.gif.zip](https://github.com/bbrown949/code-journal/files/11298405/ezgif.com-optimize.gif.zip)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -45,10 +46,10 @@ <h2 class="new-entry">New Entry</h2>
           </div>
           <div class="column-full">
             <div>
-              <label for="notes">Notes</label>
+              <div class="notes-label"><label for="notes-id">Notes</label></div>
```
Comment (shawnkost) [inline index.html]: The `for` attribute should just be `notes`. It needs to match whatever the `id` value is

File: js/main.js [hunk 2]
```diff
@@ -4,32 +4,58 @@ const $imagePlaceholder = document.querySelector('.image-placeholder');
...
+// bs FINE NOW
+const $idTitle = document.getElementById('new-entry-edit-entry');
 
 $photoURL.addEventListener('input', function (event) {
   $imagePlaceholder.setAttribute('src', event.target.value);
 });
 
-const $entryForm = document.querySelector('form');
-$entryForm.addEventListener('submit', function (event) {
+// const $entryForm = document.querySelector('form');
```
Comment (shawnkost) [inline js/main.js]: Delete all old code!

File: js/main.js [hunk 3]
```diff
@@ -4,32 +4,58 @@ const $imagePlaceholder = document.querySelector('.image-placeholder');
...
+      comment: $form.elements.notes.value,
+      url: $form.elements.url.value,
+      entryId: data.nextEntryId
+    };
+    data.nextEntryId++;
+    data.entries.unshift(singleData);
+    $imagePlaceholder.setAttribute('src', entryImagePlaceholder);
+    $form.reset();
+
+    // const $renderEntry = renderEntry(singleData);
```
Comment (shawnkost) [inline js/main.js]: Delete all old code

File: js/main.js [hunk 4]
```diff
@@ -83,17 +117,28 @@ function viewSwap(view) {
...
+  $entryFormText.textContent = 'New Entry';
+  data.editing = null;
   viewSwap('entries');
+
 });
 
 const $newButton = document.querySelector('.new-btn');
 $newButton.addEventListener('click', function () {
+
+  // $entryForm.reset();
```
Comment (shawnkost) [inline js/main.js]: Delete all old code

File: js/main.js [hunk 5]
```diff
@@ -4,32 +4,58 @@ const $imagePlaceholder = document.querySelector('.image-placeholder');
...
+    $imagePlaceholder.setAttribute('src', entryImagePlaceholder);
+    $form.reset();
+
+    // const $renderEntry = renderEntry(singleData);
+    entriesList.prepend(renderEntry(singleData));
+    viewSwap('entries');
+  } else {
+    data.editing.title = event.target.elements.title.value;
+    data.editing.comment = event.target.elements.notes.value;
+    data.editing.url = event.target.elements.url.value;
```
Comment (shawnkost) [inline js/main.js]: `data.editing` already has these values, so you shouldn't be updating `data.editing`, you want to be updating `data.entries`. So you'll need to find the current entry that exists in `data.entries` and replace it with these newly edited values

**Review by shawnkost — CHANGES_REQUESTED** [review]
Looking good so far! Just a few changes here Bryan!

**Comment by bbrown949** [comment]
updated

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**